src/pages/reset-password/[accessCode].tsx

```diff
@@ -0,0 +1,103 @@
+import { SignInButton, SignInFormItem, SignInTitle } from '@/components/auth';
+import { VerticalForm } from '@/components/common';
+import { config, showToast } from '@/lib';
+import { AuthManager } from '@/lib/managers';
+import { getMessagesFromError } from '@/lib/utils';
+import { GetServerSideProps } from 'next';
+import { useRouter } from 'next/router';
+import { SubmitHandler, useForm } from 'react-hook-form';
+import { VscLock } from 'react-icons/vsc';
+
+interface ResetPasswordProps {
+  code: string;
+}
+
+interface ResetPasswordForm {
+  newPassword: string;
+  confirmPassword: string;
+}
+
+const ResetPasswordPage = ({ code }: ResetPasswordProps) => {
+  const router = useRouter();
+
+  const {
+    register,
+    handleSubmit,
+    getValues,
+    formState: { errors },
+  } = useForm<ResetPasswordForm>();
+
+  const onSubmit: SubmitHandler<ResetPasswordForm> = ({ newPassword, confirmPassword }) => {
+    AuthManager.resetPassword({
+      user: {
+        newPassword,
+        confirmPassword,
+        code,
+      },
+      onSuccessCallback: () => {
+        showToast('Password successfully updated!');
+        router.push(config.loginRoute);
+      },
+      onFailCallback: error => {
+        showToast('Unable to reset password', getMessagesFromError(error)[0]);
+      },
+    });
+  };
+
+  return (
+    <VerticalForm onEnterPress={handleSubmit(onSubmit)}>
+      <SignInTitle text="Reset Password" />
+      <SignInFormItem
+        icon={<VscLock />}
+        type="password"
+        element="input"
+        name="password"
+        placeholder="Password"
+        formRegister={register('newPassword', {
```

Comment:
The validation logic in both `newPassword` and `confirmPassword` seem to be the same so it might make sense to create a custom validation function defined outside the component that can be used for both fields and any future password validation input boxes (create new account page for example).

Reply:
I refactored the code for validation logic across all of the existing forms we have so far and changed a decent bit of code that is unrelated so the Reset Password page so I'll open a separate PR with the changes to allow for a more isolated conversation for the larger refactor.

Reply:
https://github.com/acmucsd/membership-portal-ui-v2/pull/17